modules/mono/glue/GodotSharp/GodotSharp/Core/Mathf.cs

```diff
@@ -1298,6 +1298,18 @@ public static float MoveToward(float from, float to, float delta)
             return from + (Math.Sign(to - from) * delta);
         }
 
+        /// <summary>
+        /// Moves <paramref name="from"/> toward <paramref name="to"/> smoothly (fast to slow) by the <paramref name="delta"/> amount. The function is defined as <code>lerp(from, to, 1 - exp(-delta))</code>.
+        /// </summary>
+        /// <param name="from">The start value.</param>
+        /// <param name="to">The value to move towards.</param>
+        /// <param name="delta">The amount to move by.</param>
+        /// <returns>The value after moving smoothly.</returns>
+        public static float MoveTowardSmooth(float from, float to, float delta)
+        {
+            return Lerp(from, to, -ExpM1(-delta));
```

Comment:
`ExpM1` is not defined here. I believe starting with .NET7 it exists as `float.ExpM1()` and `double.ExpM1()`. But we are still targeting .NET6, so we need to define the method ourselves in our `Mathf` static class.

Reply:
I'm currently building with mono to finally test that but it seems I have to rely on `.NET8` because `modules/mono/global.json` seems to have a requirement on that. Am I missing something?

Reply:
> I'm currently building with mono to finally test that but it seems I have to rely on `.NET8` because `modules/mono/global.json` seems to have a requirement on that. Am I missing something?

We build using SDK8 (that is what `modules/mono/global.json` is for), but targeting .NET6. So no, there is nothing you are missing 🙂 

> Also, for some reason, I had to explicit `this.Lerp()` instead of just `Lerp()` in the `Vector2` and `Vector3` classes to prevent a warning about "this" not being used (and thus could be a static). I'm not sure which `Lerp()` was called since the function requires 3 arguments, and the method definitely requires "this". I'm not familiar with C# at all so I might be missing something obvious again.

It should not be necessary. I checked out your branch, and can't see that warning (even without `this.`).

Sidenote @raulsntos, is `AggressiveInlining` necessary above? I've been wondering recently if we should not do a pass on those, I feel like a lot of them are overzealous.

Reply:
Yeah I'm not sure now I don't see the warning anymore; might have been due to a cascading error earlier. `this` removed.

Reply:
> is `AggressiveInlining` necessary above? I've been wondering recently if we should not do a pass on those, I feel like a lot of them are overzealous.

I'm not sure, we could probably rely on the JIT doing the right thing. I'm pretty sure we just do `AggressiveInlining` because that's what `System.Math` does too.

Reply:
So what's the resolution for this thread?

Reply:
> So what's the resolution for this thread?

`ExpM1` has been implemented in the first commit, so as far as I know this thread can be marked as resolved, but I'm leaving that to the thread opener to decide.